### PR TITLE
ENH: Support nullable integer data type in to_file

### PR DESCRIPTION
**Pull request by martinfleis**
Closes #1200 

This adds support of pandas Int64 (nullable integer dtype) to `infer_schema`, allowing GeoDataFrame with Int64 column to be saved to file.

I am closing #1200 which meant to carry the whole schema and reuse it on save, but this proved to be tricky as it is extremely sensitive to any change of gdf.

### CODE REVIEW

File: geopandas/io/file.py [hunk 1]
```diff
@@ -123,17 +123,16 @@ def to_file(df, filename, driver="ESRI Shapefile", schema=None, **kwargs):
...
+    from collections import OrderedDict
 
     def convert_type(column, in_type):
         if in_type == object:
             return "str"
         if in_type.name.startswith("datetime64"):
             # numpy datetime type regardless of frequency
             return "datetime"
+        if in_type == "Int64":
+            return "int"
```
Comment (jorisvandenbossche) [inline geopandas/io/file.py]: We can probably also add a "string" -> "str" conversion for the new string dtype (only in master), and actually also "boolean" -> "bool" (that's only in a PR). 

So maybe we should keep a dictionary of conversions for those?

Comment (martinfleis) [reply geopandas/io/file.py]: Make sense. There might be more in the future, dict will make it expandable. Will do.

**Comment by martinfleis** [comment]
@jorisvandenbossche Changed per comments.

And it looks that brand new rtree 0.9 broke our tests.

**Comment by jorisvandenbossche** [comment]
Thanks for the updates!

**Pull request merged**